compiler/rustc_hir_typeck/src/cast.rs

```diff
@@ -140,7 +140,10 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             | ty::Never
             | ty::Dynamic(_, _, ty::DynStar)
             | ty::Error(_) => {
```

Comment:
You don't even need `span_delayed_bug` I think, the ty::Error carries an `ErrorGuaranteed` you can retirn here

Reply:
The type in question, which is `t` here, may not always by a `ty::Error()`. It can be any of the other varian listed in this `match` arm.

The error in the type is only found during WF check and once it is emitted as a diagnostic it is lost. The above code runs later during typeck and at that point the type actually may not carry any error.

Reply:
Oh oops I had not noticed the or-patterns my bad